msal/application.py

```diff
@@ -247,6 +248,9 @@ def get_authorization_request_url(
             You will have to specify a value explicitly.
             Its valid values are defined in Open ID Connect specs
             https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest
+        :param nonce:
+            A hard-to-guess string used to mitigate replay attacks. See also
```

Comment:
>hard-to-guess string [](start = 14, length = 20)

I suggest being a bit more specific here on the min complexity suggested

Reply:
We can write it as : `unguessable  value (ex : cryptographic hash of a random value)` instead of hard-to-guess as mentioned in the [OIDC Specs](https://openid.net/specs/openid-connect-core-1_0.html#NonceNotes).

Reply:
Yes, that paragraph was what I referred to as "another full paragraph as implementation notes" in my previous message. Did not use that "cryptographic hash of the [cryptographically random] value" part because, albeit it sounds cool, but a "cryptographically random value" is already unguessable, and its hash does not make it more unguessable. That hash suggestion, in particular, is more for a web server client, yet our MSAL does not solely target web clients, so that part is not always applicable.

Taking everything in this thread into consideration, I think we can combine all our input here, and change that line to:

    A cryptographically random value used to mitigate replay attacks. See also (that link).

I will send out a new commit soon.